File: RssReader/app/src/main/java/com/example/sweethome/rssreader/screens/settings/view/SettingsActivity.java
package com.example.sweethome.rssreader.screens.settings.view;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.example.sweethome.rssreader.R;

public final class SettingsActivity extends AppCompatActivity {
    private SettingsView mSettingsView;

    public static Intent newIntent(final Context context) {
        return new Intent(context, SettingsActivity.class);
    }

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        mSettingsView = new SettingsView(this);
        mSettingsView.onCreate();
    }

    @Override
    protected void onResume() {
        super.onResume();
        mSettingsView.onResume(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        mSettingsView.onPause();
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: HRHelper/app/src/main/java/com/example/sweethome/hrhelper/presentation/screens/settings/view/SettingsActivity.kt
package com.example.sweethome.hrhelper.presentation.screens.settings.view

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.example.sweethome.hrhelper.R
import com.example.sweethome.hrhelper.domain.entity.Event

class SettingsActivity : AppCompatActivity() {
    private val KEY_CURRENT_EVENT = "current_event"
    private lateinit var mySettingsActivityView: SettingsActivityView
    private lateinit var myEvent: Event

    companion object {
        fun newIntent(context: Context?): Intent {
            return Intent(context, SettingsActivity::class.java)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        myEvent = intent.getParcelableExtra(KEY_CURRENT_EVENT)
        mySettingsActivityView =
                SettingsActivityView(this, myEvent)
        mySettingsActivityView.onCreate()
    }

    override fun onResume() {
        super.onResume()
        mySettingsActivityView.onResume(this)
    }

    override fun onPause() {
        super.onPause()
        mySettingsActivityView.onPause()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return super.onSupportNavigateUp()
    }
}
